emu/src/render/lcd.rs

```diff
@@ -0,0 +1,41 @@
+use super::{color::colors, color::Color, DISPLAY_HEIGHT, DISPLAY_WIDTH};
+
+pub struct Lcd {
+    pixels: [Color; DISPLAY_WIDTH * DISPLAY_HEIGHT],
+}
+
+impl Lcd {
+    pub const fn new() -> Self {
+        Self {
+            pixels: [colors::BLACK; DISPLAY_WIDTH * DISPLAY_HEIGHT],
+        }
+    }
+
+    pub fn set_pixel(&mut self, x: usize, y: usize, color: Color) {
+        self[(x, y)] = color;
+    }
+}
+
+impl std::ops::Index<(usize, usize)> for Lcd {
+    type Output = Color;
+
+    fn index(&self, (x, y): (usize, usize)) -> &Color {
+        assert!(x < DISPLAY_WIDTH && y < DISPLAY_HEIGHT);
+        &self.pixels[y * DISPLAY_WIDTH + x]
```

Comment:
`x` are rows and `y` columns? If it is so, shouldn't it be `x * DISPLAY_WIDTH + y` ?

Reply:
Ok, I see below that `y` is used to indicate rows and `x` to indicate columns. So this logic is correct. Is it a convention in GBA? To me, a `row, column` access is better but I guess is subjective?

Reply:
Nop, x are the cols. Try to figure out an image in a cartesian plane.

Reply:
Yup, you're right. From an image point of view it's like this. I was thinking about matrices :D

you can solve this comment

Reply:
We can also call `col` and `row` for better explanation :D

Reply:
I'm not sure that for image (row, col) is a better description. I also check in other crates and (egui e.g.) and everywhere they use Image with x and y as this. If we move in other direction when we set the Image as egui do, we have a miss match of type in (x,y) tuple. 

I'm referring to GbaDisplay in ui when ColorImage is set as GbaLcd